includes/payment-methods/class-wc-stripe-express-checkout-ajax-handler.php

```diff
@@ -98,9 +98,7 @@ public function ajax_add_to_cart() {
 			$variation_id = $data_store->find_matching_product_variation( $product, $attributes );
 
 			WC()->cart->add_to_cart( $product->get_id(), $qty, $variation_id, $attributes );
-		}
-
-		if ( in_array( $product_type, [ 'simple', 'variation', 'subscription', 'subscription_variation', 'booking' ], true ) ) {
+		} elseif ( ! in_array( $product_type, [ 'grouped', 'external' ], true ) ) { // Grouped and external products should not be added to the cart.
```

Comment:
Instead of allowing any `product_type`, I think we should use the `supported_product_types()` function, which allows the list to be filtered by `wc_stripe_payment_request_supported_types` as mentioned in the issue.

https://github.com/woocommerce/woocommerce-gateway-stripe/blob/7779798d9031963026f18c09dbbfd428641320ab/includes/payment-methods/class-wc-stripe-express-checkout-helper.php#L343-L358

Reply:
Alright! I think it works, too. I have updated the main implementation to use these methods instead 👍